src/UseCases/GetInTouch/GetInTouchUseCase.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace WMDE\Fundraising\Frontend\UseCases\GetInTouch;
+
+use WMDE\Fundraising\Frontend\MailAddress;
+use WMDE\Fundraising\Frontend\Messenger;
+use WMDE\Fundraising\Frontend\Validation\GetInTouchValidator;
+
+/**
+ * @license GNU GPL v2+
+ * @author Kai Nissen < kai.nissen@wikimedia.de >
+ */
+class GetInTouchUseCase {
+
+	private $validator;
+	private $messenger;
+	/** @var GetInTouchRequest */
+	private $request;
+
+	public function __construct( GetInTouchValidator $validator, Messenger $messenger ) {
+		$this->validator = $validator;
+		$this->messenger = $messenger;
+	}
+
+	public function processContact( GetInTouchRequest $request ): string {
+		$this->request = $request;
+
+		if ( !$this->validator->validate( $request ) ) {
+			return 'validation failed';
+		}
+
+		if ( $this->forwardContactRequest() && $this->confirmToUser() ) {
+			return 'request successful';
+		}
+
+		return 'mail transmission failed';
+	}
+
+	private function forwardContactRequest(): bool {
+		$this->messenger->sendMessage(
+			$this->messenger->constructMessage(
+				new MailAddress(
+					$this->request->getEmailAddress(),
+					implode( ' ', [ $this->request->getFirstName(), $this->request->getLastName() ] )
+				),
+				new MailAddress( 'kai.nissen@wikimedia.de' ),
+				$this->request->getSubject(),
+				$this->request->getMessageBody()
+			)
+		);
+
+		return $this->messenger->getFailedRecipients() === [];
```

Comment:
This code looks like it deals with multiple recipients but it does not correctly I think. As soon as there is one recipient to which the message was not send, this code will behave as if sending altogether failed, while stuff was still send to the other recipients.

`Messenger` currently only sends to a single one, so perhaps its interface should just be changed accordingly. Rather than having this `getFailedRecipients`, the `sendMessage` method could throw an exception on failure.